mlflow/store/_unity_catalog/registry/uc_oss_rest_store.py

```diff
@@ -0,0 +1,468 @@
+
+_METHOD_TO_INFO = extract_api_info_for_service(UnityCatalogService, _UC_OSS_REST_API_PATH_PREFIX)
+_METHOD_TO_ALL_INFO = extract_all_api_info_for_service(
+    UnityCatalogService, _UC_OSS_REST_API_PATH_PREFIX
+)
+
+def _raise_unsupported_arg(arg_name, message=None):
+        messages = [
+            f"Argument '{arg_name}' is unsupported for models in the Unity Catalog.",
+        ]
+        if message is not None:
+            messages.append(message)
+        raise MlflowException(" ".join(messages))
+    
+def _require_arg_unspecified(arg_name, arg_value, default_values=None, message=None):
+    default_values = [None] if default_values is None else default_values
+    if arg_value not in default_values:
+        _raise_unsupported_arg(arg_name, message)
+
+
+@experimental
+class UnityCatalogOssStore(BaseRestStore):
+    """
+    Client for an Open Source Unity Catalog Server accessed via REST API calls.
+    """
+
+    def __init__(self, store_uri):
+        super().__init__(get_host_creds=functools.partial(get_oss_host_creds, store_uri))
+        self.tracking_uri = None  # OSS has no tracking URI
+
+    def _get_response_from_method(self, method):
+        method_to_response = {
+            CreateRegisteredModel: RegisteredModelInfo,
+            CreateModelVersion: ModelVersionInfo,
+            UpdateRegisteredModel: RegisteredModelInfo,
+            DeleteRegisteredModel: DeleteRegisteredModel,
+            DeleteModelVersion: DeleteModelVersion.Response,
+            GetRegisteredModel: RegisteredModelInfo,
+            GetModelVersion: ModelVersionInfo,
+            FinalizeModelVersion: ModelVersionInfo,
+            UpdateModelVersion: ModelVersionInfo,
+            GenerateTemporaryModelVersionCredential: TemporaryCredentials,
+            ListRegisteredModels: ListRegisteredModels.Response,
+            ListModelVersions: ListModelVersions.Response,
+        }
+        return method_to_response[method]()
+
+    def _get_endpoint_from_method(self, method):
+        return _METHOD_TO_INFO[method]
+
+    def _get_all_endpoints_from_method(self, method):
+        return _METHOD_TO_ALL_INFO[method]
+
+    def create_registered_model(self, name, tags=None, description=None):
+        full_name = get_full_name_from_sc(name, None)
+        [catalog_name, schema_name, model_name] = full_name.split(".")
+        comment = description if description else ""
+        # RegisteredModelInfo is inlined in the request and the response.
+        # https://docs.databricks.com/api/workspace/registeredmodels/create
+        req_body = message_to_json(
+            CreateRegisteredModel(
+                name=model_name,
+                catalog_name=catalog_name,
+                schema_name=schema_name,
+                comment=comment,
+            )
+        )
+        registered_model_info = self._call_endpoint(CreateRegisteredModel, req_body)
+        return registered_model_from_uc_oss_proto(registered_model_info)
```

Comment:
Would it be possible to rename this util (and the other util functions in https://github.com/mlflow/mlflow/blob/af8ffc7c350053c314fca459038a10dcdb72e390/mlflow/utils/_unity_catalog_oss_utils.py) to use imperative naming?

Reply:
Done